test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,11 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+
+        if(inputCommand.equals("add") || inputCommand.equals("clear") || inputCommand.equals("delete")) {
```

Comment:
Try to check the command type using a existing class elegantly rather than using "equals". Meanwhile, test the else branch too, i.e. when the command is not mutating.Think about what should be tested and how to test it.